src/lib/evil/evil_stat.c

```diff
@@ -0,0 +1,126 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "evil_private.h"
+#include <evil_api.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include "evil_stat.h"
+
+#define	AT_SYMLINK_NOFOLLOW	0x01
+
+EAPI int 
+fstatat(int dirfd, const char *pathname, struct stat *statbuf, int flags)
+{
+   int r_fstatat;
+
+   if (pathname[2] == '\\')
```

Comment:
What if I pass the path `ab\`?
Moreover, we should validate the `pathname` length before accessing. if `pathname` contains `"a"`, we hit a buffer overrun.

Reply:
And what if the path separator is `/`?

Reply:
if the separator is: "/" or the path is "ab \", simply skip this step, and skip to the next step.

Reply:
If the path starts with `C:/`, shouldn't it enter the `if`? And if the path is `ab\`, it will enter the `if`, but if it is `abc\`, it won't. Is this the expected behavior?

Reply:
If the path has a "/" in the third character (windows), I understand that the user gave me the full path; in this case, it enters the if.
caso contrário, você entenderá quem é o usuário ou o software que está passando um nome de arquivo ou diretorio, ai não entra no if.
if the user passes paths, such as: "ab / img.jpg" or "abc / img.jpg", it will return -1 entering or not in the if, because it is not a directory name, it is not a file name and not a full file path

Reply:
"ab/" is a relative directory name. Furthermore, what if the path is "\Windows"? It is an absolute path, will the function work?

Reply:
In this case, the function returns -1, because a name was passed incorrectly

Reply:
Why is it incorrect?

Reply:
In fact, I expressed myself poorly

I will quote an example: I want to know the size of a given file and, for that, I will use fstatat ():

1. </test/test.txt> in this case, if the file exists and the relative directory exists, it will work correctly.

2. </test/> in this case, if the relative directory exists, it will work correctly and you can get the size of the directory.

3. </test/> in this case, if the relative directory does not exist, the function returns -1.

In this first part of the function, I did just to identify absolute paths.

Reply:
> In this first part of the function, I did just to identify absolute paths.

The directory `ab\` is a relative path, but the way the code is, it will identify it as an absolute path.